### DISCUSSION TITLE
Pengurusan Qurban

### DISCUSSION THREAD

**Discussion by rohaizanr** [Dev4w4n/e-masjid.my#128] [Ideas]
Lembu 7 bahagian
Cara public untuk booking/confirm/bayaran ?
Notification ?

**Comment by AimanFariz** [comment]
Saya cadangkan kita gunakan konsep scan QR code seperti modul 'Cadangan' untuk public buat booking korban/aqiqah.

Cadangan untuk QR Code tersebut:
1. QR Code akan bawa ke page booking yang berada di bawah domain e-masjid, routenya mungkin akan jadi e-masjid.my/tempahan atau apa2 nama yang sesuai.
2. Pengguna (Public) boleh pilih nak buat tempahan korban atau aqiqah, kemudian boleh tambah penyertaan.
3. Selepas tambah penyertaan, pengguna boleh confirmkan order -> dibawa ke page seterusnya untuk isi maklumat dan lengkapkan butiran bayaran.

Ini sedikit lo-fi prototype yang saya cadangkan:
https://github.com/Dev4w4n/e-masjid.my/assets/99475959/1c664ada-0864-4752-a878-66efa665bd49

Dekat pop-up untuk buat tempahan tu, pengguna boleh pilih jenis haiwan: Kambing/Lembu
Dekat Kuantiti tu, boleh pilih dari nilai 1 hingga 7 (maximum) kalau tempahan Korban.
Bila pengguna click 'Simpan & Tutup' atau 'Simpan & Tambah Peserta', nanti tempahan yang mereka buat tadi akan diupdate dalam tempahan sedia ada.
Jumlah (RM) akan diupdate juga, (kuantiti bahagian * harga bagi satu bahagian).

**Reply by imie** [reply]
cadangan yang baik tuan. dan juga ada tambahan cadangan dari saya. ajk boleh preset berapa ekor lembu/kambing mereka nak open. mungkin kerana kekangan masa atau tempat atau tenaga kerja. boleh jadi mereka hanya buat korban maksimum beberapa ekor sahaja. 

untuk preset, kita listkan berapa ekor lembu/kambing yang available dan berapa slot yang telah diambil.

**Comment by rohaizanr** [comment]
terima kasih @AimanFariz & @imie 

Ini adalah antara modul yang kurang 'linkage' dengan modul lain, agak stand-alone.

Sesuai juga untuk dibangunkan memandangkan, Raya Qurban tak lama lagi dah

**Comment by AimanFariz** [comment]
Baik. Saya akan bantu development dari segi design UI.

Ini sambungan daripada cadangan saya di atas ya.

Apabila public tekan butang 'Simpan & Tutup' atau 'Simpan dan Tambah Peserta', info yang dimasukkan oleh public tadi akan dimasukkan dalam penyertaan terkini:
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/e8d6a3da-e517-49cc-ab6b-6067d7be1ec6)

Tambahan, pengguna akan ada option untuk kemas kini tempahan atau buang tempahan. Kalau pengguna click 'Kemas Kini Tempahan', window tempahan akan dibuka semula, dengan data daripada tempahan tersebut.
Kalau pengguna click 'Buang Tempahan', popup window untuk confirmation akan tanya sama ada pengguna betul2 nak buang tempahan akan muncul. Kalau pengguna click 'Ya', tempahan tersebut akan dibuang.

Apabila public tutup window tempahan, mereka akan nampak option untuk melakukan akad, contohnya seperti ini:
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/427ca7cd-90e2-4633-9e03-ce960f0978e3)

Seterusnya adalah butang Akad. Apabila pengguna sudah selesai memasukkan details tempahan, mereka akan klik butang 'Akad' dan dibawa ke page seterusnya:
![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/b5b8ce8f-a900-4b5f-9acb-9068ccacdb43)

Disclaimer: Saya mengambil inspirasi daripada laman web EZ Qurban untuk design UI ini. Mereka asingkan page 'Maklumat' dan 'Akad', tapi saya rasa kedua-dua page ni boleh disatukan saja.
Satu lagi, 'Jumlah kecil' merujuk kepada jumlah harga untuk setiap unit tempahan, 'Jumlah' merujuk kepada jumlah harga untuk semua unit.

Tambahan:
Untuk jenis pengenalan diri, pengguna boleh pilih sama ada no. mykad/no tentera / nombor passport (kes bukan warganegara)

Untuk nombor telefon,
pengguna boleh pilih negara, dan kod telefon negara akan automatic sync dengan flag negara tersebut. (Contoh: +60 untuk Malaysia, +1 untuk USA)

**Reply by rohaizanr** [reply]
nice ni @AimanFariz
